File: deploy.py
import paramiko
import os
from dotenv import load_dotenv
import time

load_dotenv()

def deploy():
    host = os.getenv('SFTP_HOST')
    port = int(os.getenv('SFTP_PORT'))
    user = os.getenv('SFTP_USER')
    password = os.getenv('SFTP_PASS')
    local_dir = 'out'

    print(f"🚀 Iniciando despliegue en {host} (Modo Compatibilidad)...")
    
    try:
        # Forzar algoritmos que GoDaddy suele aceptar
        ssh = paramiko.SSHClient()
        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        
        # Ajustar algoritmos para evitar EOFError
        ssh.connect(
            host, 
            port=port, 
            username=user, 
            password=password, 
            look_for_keys=False, 
            allow_agent=False,
            disabled_algorithms={'pubkeys': ['rsa-sha2-256', 'rsa-sha2-512']}
        )
        
        sftp = ssh.open_sftp()
        print("✅ Conexión SFTP establecida con éxito.")
        
        # Navegar al directorio correcto
        try:
            sftp.chdir('html')
            print("📁 Directorio remoto: /html")
        except:
            print("📁 Directorio remoto: /")

        def upload_dir(local_path, remote_subpath):
            files = os.listdir(local_path)
            for item in files:
                l_path = os.path.join(local_path, item)
                r_path = (remote_subpath + '/' + item).strip('/')
                
                if os.path.isdir(l_path):
                    try:
                        sftp.mkdir(r_path)
                    except:
                        pass
                    upload_dir(l_path, r_path)
                else:
                    print(f"📤 Subiendo: {r_path}")
                    sftp.put(l_path, r_path)

        upload_dir(local_dir, '')
        
        sftp.close()
        ssh.close()
        print("🎉 ¡TODO SUBIDO! La web de Limpia MAX ya debería estar online.")
        
    except Exception as e:
        import traceback
        print(f"❌ Error: {e}")
        traceback.print_exc()

if __name__ == "__main__":
    deploy()

File: deploy.mjs
import sftp from 'ssh2-sftp-client';
import path from 'path';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const config = {
  host: process.env.SFTP_HOST || 'r5l.550.myftpupload.com',
  port: parseInt(process.env.SFTP_PORT || '22'),
  username: process.env.SFTP_USER || 'client_7099bf3c0_1103883',
  password: process.env.SFTP_PASSWORD || 'WM4jgbMFQuQ0UM',
  privateKey: process.env.SFTP_KEY,
  retries: 3,
  retry_factor: 2,
  retry_minTimeout: 2000
};

// Si hay una clave privada en el archivo local y no viene por env, cargarla (para uso local)
if (!config.privateKey && !config.password) {
    try {
        const keyPath = path.join(__dirname, 'limpiamax_key');
        if (fs.existsSync(keyPath)) {
            config.privateKey = fs.readFileSync(keyPath);
        }
    } catch {
        console.log('ℹ️ No se encontró clave privada local, usando contraseña.');
    }
}

const client = new sftp();

async function deploy() {
  const localDir = path.join(__dirname, 'out');
  const remoteDir = '/html'; // Use absolute path to bypass symlink issues

  try {
    console.log('🚀 Iniciando despliegue de Limpia MAX...');
    console.log(`📂 Conectando a ${config.host} como ${config.username}...`);
    
    await client.connect(config);
    console.log('✅ Conexión SFTP establecida con éxito.');

    // Rename existing index.php if it exists to allow index.html to take priority
    try {
      if (await client.exists('./html/index.php')) {
        console.log('🔄 Renombrando index.php antiguo de WordPress...');
        await client.rename('./html/index.php', './html/index.php.bak');
      }
    } catch {
      console.log('⚠️ No se pudo renombrar index.php (posiblemente ya no existe).');
    }

    console.log(`📤 Subiendo contenido de ${localDir} a ${remoteDir}...`);
    
    // Upload the entire directory recursively
    await client.uploadDir(localDir, remoteDir);

    console.log('✨ ¡Despliegue completado con éxito!');
  } catch (err) {
    console.error('❌ Error durante el despliegue:', err.message);
    if (err.message.includes('authentication')) {
      console.log('💡 Sugerencia: Asegúrate de que la clave SSH haya sido vinculada correctamente en el panel de GoDaddy.');
    }
  } finally {
    await client.end();
  }
}

deploy();
